Framework/PythonInterface/core/inc/MantidPythonInterface/core/Copyable.h

```diff
@@ -0,0 +1,55 @@
+
+// Mantid Repository : https://github.com/mantidproject/mantid
+//
+// Copyright &copy; 2019 ISIS Rutherford Appleton Laboratory UKRI,
+//     NScD Oak Ridge National Laboratory, European Spallation Source
+//     & Institut Laue - Langevin
+// SPDX - License - Identifier: GPL - 3.0 +
+#ifndef MANTID_PYTHONINTERFACE_COPYABLE_H_
+#define MANTID_PYTHONINTERFACE_COPYABLE_H_
+#include <boost/python/class.hpp>
+#include <boost/python/dict.hpp>
+#include <boost/python/extract.hpp>
+#include <boost/python/import.hpp>
+#include <boost/python/manage_new_object.hpp>
+namespace Mantid {
+namespace PythonInterface {
+
+// Implementation based on
+// https://mail.python.org/pipermail/cplusplus-sig/2009-May/014505.html
+namespace bp = boost::python;
+template <typename T> inline PyObject *managingPyObject(T *p) {
```

Comment:
This looks great and useful for other types. Could we add some function documentation strings just to be clear about the purpose of each function?